### ISSUE TITLE
scipy module error?

### ISSUE THREAD

**Issue by yukaribbba** [pytroll/satpy#1835]
**Describe the bug**
`FileNotFoundError: Could not find module 'C:\Users\45107\anaconda3\lib\site-packages\scipy\.libs\libbanded5x.UGR6EUQPIWHQH7SL62IWIXB5545VDNQZ.gfortran-win_amd64.dll' (or one of its dependencies). Try using the full path with constructor syntax.`
Meanwhile, that `libbanded5x.UGR6EUQPIWHQH7SL62IWIXB5545VDNQZ.gfortran-win_amd64.dll` is actually in the right place.

**To Reproduce**

```python
# Any satpy script seems to result in this error.

```

**Actual results**
```
Traceback (most recent call last):
  File "C:\Users\45107\Downloads\Sat\s2t.py", line 2, in <module>
    from satpy import Scene, find_files_and_readers
  File "C:\Users\45107\anaconda3\lib\site-packages\satpy\__init__.py", line 25, in <module>
    from satpy.utils import get_logger  # noqa
  File "C:\Users\45107\anaconda3\lib\site-packages\satpy\utils.py", line 30, in <module>
    import xarray as xr
  File "C:\Users\45107\anaconda3\lib\site-packages\xarray\__init__.py", line 3, in <module>
    from . import testing, tutorial, ufuncs
  File "C:\Users\45107\anaconda3\lib\site-packages\xarray\testing.py", line 8, in <module>
    from xarray.core import duck_array_ops, formatting, utils
  File "C:\Users\45107\anaconda3\lib\site-packages\xarray\core\duck_array_ops.py", line 15, in <module>
    from . import dask_array_compat, dask_array_ops, dtypes, npcompat, nputils
  File "C:\Users\45107\anaconda3\lib\site-packages\xarray\core\dask_array_compat.py", line 5, in <module>
    from .pycompat import dask_version
  File "C:\Users\45107\anaconda3\lib\site-packages\xarray\core\pycompat.py", line 55, in <module>
    dsk = DuckArrayModule("dask")
  File "C:\Users\45107\anaconda3\lib\site-packages\xarray\core\pycompat.py", line 25, in __init__
    duck_array_type = (import_module("dask.array").Array,)
  File "C:\Users\45107\anaconda3\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "C:\Users\45107\anaconda3\lib\site-packages\dask\array\__init__.py", line 3, in <module>
    from . import backends, fft, lib, linalg, ma, overlap, random
  File "C:\Users\45107\anaconda3\lib\site-packages\dask\array\backends.py", line 13, in <module>
    from .percentile import _percentile
  File "C:\Users\45107\anaconda3\lib\site-packages\dask\array\percentile.py", line 10, in <module>
    from .core import Array
  File "C:\Users\45107\anaconda3\lib\site-packages\dask\array\core.py", line 64, in <module>
    from .chunk_types import is_valid_array_chunk, is_valid_chunk_type
  File "C:\Users\45107\anaconda3\lib\site-packages\dask\array\chunk_types.py", line 129, in <module>
    import scipy.sparse
  File "C:\Users\45107\anaconda3\lib\site-packages\scipy\__init__.py", line 136, in <module>
    from . import _distributor_init
  File "C:\Users\45107\anaconda3\lib\site-packages\scipy\_distributor_init.py", line 59, in <module>
    WinDLL(os.path.abspath(filename))
  File "C:\Users\45107\anaconda3\lib\ctypes\__init__.py", line 373, in __init__
    self._handle = _dlopen(self._name, mode)
FileNotFoundError: Could not find module 'C:\Users\45107\anaconda3\lib\site-packages\scipy\.libs\libbanded5x.UGR6EUQPIWHQH7SL62IWIXB5545VDNQZ.gfortran-win_amd64.dll' (or one of its dependencies). Try using the full path with constructor syntax.
```

**Environment Info:**
 - satpy: 0.30
 - scipy: 1.7.1

**Comment by djhoese** [comment]
If you do `python -c "import scipy.sparse"` do you get a similar error?

How did you install scipy? This is very likely nothing to do with satpy and more likely an issue with how you installed scipy.

**Comment by yukaribbba** [comment]
Yup, the same one. I didn't change anything since `conda install -c conda-forge satpy --all`.

**Comment by djhoese** [comment]
You may need to try recreating your environment. Doing `--all` on the install command updates **all** packages in your environment. This makes it very easy to put packages into an incompatible state. You could also try running `conda update --all` again and see if updating everything fixes things.

This is not a satpy issue. It seems your conda environment has become corrupt/inconsistent with the other packages (ex. scipy and fortran compiler, but this probably isn't the issue here, just an example). That python command I had you run doesn't touch satpy at all.

**Comment by yukaribbba** [comment]
I reinstalled everything without `--all` option and problem got solved. Thanks!

**Comment by jasma-balasangameshwara** [comment]
Hi, I am getting the same error!
FileNotFoundError: Could not find module 'C:\...\lib\site-packages\scipy\.libs\libbanded5x.UGR6EUQPIWHQH7SL62IWIXB5545VDNQZ.gfortran-win_amd64.dll' (or one of its dependencies). Try using the full path with constructor syntax. How to I fix it? @yukaribbba , please elaborate on 'I reinstalled everything without --all option and problem got solved. Thanks!'.

**Comment by yukaribbba** [comment]
@jasma-balasangameshwara Well this error occurred three months ago. At that time, my solution was switching scipy from **conda-forge channel** to **pkgs/main channel** by `conda install -c defaults scipy`. Didn't know why but it worked. Also, it seems they have fixed this issue in the latest scipy 1.7.3. Now I'm using this version from **conda-forge** and nothing is wrong here.

**Comment by jasma-balasangameshwara** [comment]
@yukaribbba, the issue was resolved. Thank you!

**Comment by Bargus11** [comment]
Vraiment merci, la commande '' conda install -c defaults scipy '' a marché a merveille

elle a effacée tous les fichiers dll présents dans \site-package\scipy\.libs et a crée un seul fichier équivalent nommé libansari.EOFVGRJDNFSBN5SEHMGYDAIAVH4RFCJY.gfortran-win_amd64.dll.c~conda_trash   

Au passage mon probleme etait le suivant : quand je compilais la commande suivante  "from sklearn.datasets import make_blobs"  j'obtenais cette erreur
"Could not find module 'C:\ProgramData\Anaconda3\lib\site-packages\scipy\.libs\libbanded5x.CBCBXIAYFUWOYCSBXUA2BT6W76FYX4C7.gfortran-win_amd64.dll' (or one of its dependencies). Try using the full path with constructor syntax."

**Comment by jasma-balasangameshwara** [comment]
Yes, after FileNotFound error was resolved using 'conda install -c
Vonda-forge scipy ', I got ModuleNotFound error. I am using Pycharm IDE, I
installed scipy package manually on Pycharm. The issue was resolved.

**Comment by eyildiz-ugoe** [comment]
The issue is still there if you install scikit with pip on Anaconda.

**Comment by djhoese** [comment]
@eyildiz-ugoe This is not a satpy issue. This is a scipy issue.

That said, it is generally a bad idea to install python packages that contain compiled extensions (scipy, rasterio, pyproj) with pip into a conda environment with other low-level packages. It is also usually a bad idea to mix packages from the default Anaconda channel and conda-forge (I know someone above had luck with that but still).

Bottom line, please file bug reports with scipy. There isn't anything we (the satpy community) can do to help.
